File: products-microservice/src/models/Product.go
package models

import "gorm.io/gorm"

type Product struct {
	gorm.Model
	Name     string `gorm:"type:varchar(100);unique;not null"`
	Category string `gorm:"type:varchar(100);not null"`
	Price    int    `gorm:"type:int;not null"`
	Stock    int    `gorm:"type:int;not null"`
}

File: sales-microservice/src/models/Product.ts
export class Product {
    productID: number;
    name: string;
    price: number;
    quantity: number;

    constructor(productID: number, name: string, price: number, quantity: number) {
        this.productID = productID;
        this.name = name;
        this.price = price;
        this.quantity = quantity;
    }
}